components/requestreply/backend_pubsub.go

```diff
@@ -0,0 +1,278 @@
+package requestreply
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+// PubSubBackend is a Backend that uses Pub/Sub to transport commands and replies.
+type PubSubBackend[Result any] struct {
+	config    PubSubBackendConfig
+	marshaler BackendPubsubMarshaler[Result]
+}
+
+// NewPubSubBackend creates a new PubSubBackend.
+//
+// If you want to use backend together with `NewCommandHandler` (without result), you should pass `NoResult` or `struct{}` as Result type.
+func NewPubSubBackend[Result any](
+	config PubSubBackendConfig,
+	marshaler BackendPubsubMarshaler[Result],
```

Comment:
it's generic and it would force config to depend on generic type - in other words you  couldn't have a single confg instance that could be used for multiple Result types